### DISCUSSION TITLE
v4.0.26.0

### DISCUSSION THREAD

**Discussion by doug24** [dnGrep/dnGrep#877] [General]
[![v4.0.26.0](https://img.shields.io/github/downloads/dnGrep/dnGrep/v4.0.26.0/total.svg)](https://github.com/dnGrep/dnGrep/releases/tag/v4.0.26.0)

## Changes in this release:

**This is the first release for dnGrep running on .NET 7**  

You should see significantly faster searches.  The features and functionality should match release 3.2.330, with the addition of the issues fixed below. This release is supported only on Windows 10 and Windows 11. 

Both the msi and zip packages are self-contained, which means they include the .NET runtime with the application. The portable (zip) builds need to be self-contained, so nothing needs to be installed on the system. Making the msi packages also self-contained means there doesn't need to separate builds for msi and zip, and the user doesn't need to install the .NET runtime separately. dnGrep will be updated when Microsoft issues security patches for .NET.  This release contains the .NET v7.0.7 runtime.

#867 Hide Bookmarks Window: fixed error when Bookmarks window is closed using the titlebar 'X' or alt+F4.

#871 Update version check: changed the version check to ignore prereleases.

#872 Finding file names in archives: fixed a bug in the code matching file names in archives.

One additional performance enhancement has been added since 4.0.8-rc1 for text searches.  Thank-you to @yyjdelete for finding it.

The \Plugins\PDFSearch directory has been reorganized to put the XPDF files and pdftotext.exe in a subdirectory. This will make it easier for users to add additional language packs for processing PDF files.

**Updated Translations**  The Serbian and Italian translations were updated - thanks to the translators: @majkinetor and @sandman91 

Many of the translations still need help to complete them.  You are welcome and encouraged to complete or improve any of the existing translations, see [Translate dnGrep](https://github.com/dnGrep/dnGrep/wiki/Translate-dnGrep) for more information. Do as much or as little as you like.  If you do not want to use Weblate for translations, I will take input in any format and merge the changes.  Send me a note in the issues or discussions page.

### Checksum information

Name: dnGREP.4.0.26.x64.msi
SHA256: a1ffe5007ce57b5f229e78c60b104b42d45070c31b7ff906dbee0e9360b05e26

Name: dnGREP.4.0.26.x86.msi
SHA256: 470ce76372d999f4b4189c57663a0832f1b0ecd4501bdbe5fededb64e1c25806

Name: dnGrep.4.0.26.0.x64.zip
SHA256: d4adb24531db170ef67e45b6aca0db3c4ef2ceb69116576c7d8019f83274f6e8

Name: dnGrep.4.0.26.0.x86.zip
SHA256: 27349f1ffe85d8876cc63d5e5156b5ee74b0c6f6940040cf9ffbc067c0472730

This program uses free code signing provided by [SignPath.io](https://signpath.io?utm_source=foundation&utm_medium=github&utm_campaign=dngrep), and a certificate by the [SignPath Foundation](https://signpath.org?utm_source=foundation&utm_medium=github&utm_campaign=dngrep).

<hr /><em>This discussion was created from the release <a href='https://github.com/dnGrep/dnGrep/releases/tag/v4.0.26.0'>v4.0.26.0</a>.</em>

**Comment by fade0ff** [comment]
Hm, tried to Unzip, then to install, then to uninstall and re-install, but all I get is this (Win10):

```
<logevent time="2023-06-27 19:13:56.9909" level="Error" logger="dnGREP.WPF.App">
  <message>Failure in application startup</message>
  <exception>System.Windows.Markup.XamlParseException: The invocation of the constructor on type &apos;dnGREP.WPF.PreviewControl&apos; that matches the specified binding constraints threw an exception.
 ---&gt; System.FormatException: String &apos;AAEAAAD/////AQAAAAAAAAAEAQAAAA5TeXN0ZW0uQm9vbGVhbgEAAAAHbV92YWx1ZQABAAs=&apos; was not recognized as a valid Boolean.
   at System.Boolean.Parse(ReadOnlySpan`1 value)
   at System.String.System.IConvertible.ToBoolean(IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at dnGREP.Common.GrepSettings.Get[T](String key) in C:\projects\dngrep\dnGREP.Common\GrepApplicationSettings.cs:line 847
   at dnGREP.WPF.PreviewControl..ctor() in C:\projects\dngrep\dnGrep.WPF\Views\PreviewControl.xaml.cs:line 48
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   --- End of inner exception stack trace ---
   at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at System.Windows.Application.LoadComponent(Object component, Uri resourceLocator)
   at dnGREP.WPF.MainForm..ctor(Boolean isVisible) in C:\projects\dngrep\dnGrep.WPF\Views\MainView.xaml.cs:line 38
   at dnGREP.WPF.App.Application_Startup(Object sender, StartupEventArgs e) in C:\projects\dngrep\dnGrep.WPF\App.xaml.cs:line 63
```

**Comment by doug24** [comment]
Older versions of dnGrep put data in the user settings in binary format, which is no longer supported in .NET 7.  I put code in the last few versions to convert the binary data, but the current version cannot do that anymore. Try deleting dnGREP.Settings.dat in the %appdata%\dnGREP folder and restarting.  If you want to try to keep your other settings, edit the dnGREP.Settings.dat file and remove anything that looks like AAEAAAD/////AQAAAAAAAAAEAQAAAA5TeXN0ZW0uQm9vbGVhbgEAAAAHbV92YWx1ZQABAAs=

**Reply by fade0ff** [reply]
Ah, great, that fixed it. Thanks a lot for this super-fast and helpful reply!
Totally love this tool and use it almost every single day.
